### PR TITLE
Properly report validation errors from inline fields in strict mode

### PR DESCRIPTION
**Pull request by quentinmit**
Second half of #244...

### CODE REVIEW

**Comment by goccy** [comment]
From the next time, if it is still under development, Please use Draft PR or add WIP to the title :) .
Please contact me if you would like to review.

File: decode.go [hunk 1]
```diff
@@ -1073,6 +1068,8 @@ func (d *Decoder) decodeStruct(ctx context.Context, dst reflect.Value, src ast.N
 					}
 				}
 			}
+			// Make sure we still report errors even if we can't associate them with tokenss
```
Comment (goccy) [inline decode.go:1071]: Thanks, this comment isn't needed

File: decode.go [hunk 2]
```diff
@@ -1042,7 +1037,7 @@ func (d *Decoder) decodeStruct(ctx context.Context, dst reflect.Value, src ast.N
 		return errors.Wrapf(foundErr, "failed to decode value")
 	}
 
-	if len(unknownFields) != 0 && d.disallowUnknownField {
+	if len(unknownFields) != 0 && d.disallowUnknownField && src.GetToken() != nil {
```
Comment (goccy) [inline decode.go:1040]: Why is this added comparison ( `src.GetToken() != nil` ) necessary?

Comment (quentinmit) [reply decode.go:1040]: Before, when parsing inline structs, this block would generate unknown field errors for any fields that don't belong to the current inline struct, and then the removed block above would swallow those errors. By adding this comparison, we skip checking for unknown fields so that it can move on and perform the validation check below (since inline structs get parsed without a token).

Comment (goccy) [reply decode.go:1040]: I understood ! Thank you.

Could you please put the comment like the following ?
`If there are unkonwn fields in the embedded struct by inline, the token does not exist, so ignore this section`

File: decode.go (hunk 1 above)
Comment (quentinmit) [reply decode.go]: Okay, I removed it...

File: decode.go (hunk 2 above)
Comment (quentinmit) [reply decode.go:1042]: Done (I reworded it to try to make it even clearer).

**Comment by quentinmit** [comment]
Thank you very much for the quick reviews!

**Comment by goccy** [comment]
Thank you for the contribution ! LGTM
